### PR TITLE
Fixes #21893 Unable to start cluster if Japanese (or Non-ASCII) characters are ent…

### LINKED ISSUE #21893: Unable to start cluster if Japanese (or Non-ASCII) characters are entered in description field
## Environment Details
* GlassFish Version (and build number):glassfish 5.0 b08
* JDK version: 1.8.0_131
* OS: Windows 2012 R2 

----------

## Problem Description
Cluster server instances fail to start if the replicated domain.xml contains Japanese (or Non-ASCII) characters. 

## Steps to reproduce
The same error will be seen if the procedure is done from the command-line or from the admin GUI.

The steps listed below are for the admin GUI:
1. Navigate to Clusters > New...
2. Create a cluster "c1" with an instance "c1i1" (but do not start the cluster yet)

3. Do either of the following steps to save a non-ASCII string in the configuration:

3a. Resource string:
    - Navigate to Resources > JMS
    - Complete the form items, with "日本語名" for the Description, then click Save.

3b. Config string:
    - Navigate to Configurations > c1-config > EJB Container
    - In the Additional Properties table, add a new property "abc" with value "日本語の値" then click Save.

4. Navigate to Clusters, then select and start c1.

=> The cluster instance cannot be started. The following error is displayed at the top of the page:

An error has occurred c1i1: 
Could not start instance c1i1 on node localhost-domain1 (localhost). Command failed on node localhost-domain1 (localhost): Attempting to start c1i1.... Please look at the server log for more details..... To complete this operation run the following command locally on host localhost from the GlassFish install location C:\glassfish-5.0-b08\glassfish5: lib/nadmin start-local-instance --node localhost-domain1 --sync normal c1i1 The command start-instance failed for: c1i1 An error has occurred

=> From the command-line, the error and stacktrace are seen in the console by running "nadmin start-local-instance --node localhost-domain1 --sync normal c1i1" or "asadmin start-local-instance c1i1"

````
Caused by: java.lang.RuntimeException: Fatal Error.  Unable to parse file:/C:/gl
assfish-5.0-b08/glassfish5/glassfish/nodes/localhost-domain1/c1i1/config/domain.
xml
        at org.glassfish.config.support.DomainXml.parseDomainXml(DomainXml.java:
277)
        at org.glassfish.config.support.DomainXml.run(DomainXml.java:117)
        at org.jvnet.hk2.config.ConfigurationPopulator.populateConfig(Configurat
ionPopulator.java:58)
        at org.glassfish.hk2.bootstrap.HK2Populator.populateConfig(HK2Populator.
java:107)
        at com.sun.enterprise.module.common_impl.AbstractModulesRegistryImpl.pop
ulateConfig(AbstractModulesRegistryImpl.java:211)
        at com.sun.enterprise.module.bootstrap.Main.createServiceLocator(Main.ja
va:273)
        at org.jvnet.hk2.osgiadapter.HK2Main.createServiceLocator(HK2Main.java:1
19)
        at com.sun.enterprise.glassfish.bootstrap.osgi.EmbeddedOSGiGlassFishRunt
ime.newGlassFish(EmbeddedOSGiGlassFishRuntime.java:94)
        at com.sun.enterprise.glassfish.bootstrap.GlassFishRuntimeDecorator.newG
lassFish(GlassFishRuntimeDecorator.java:67)
        at com.sun.enterprise.glassfish.bootstrap.osgi.OSGiGlassFishRuntime.newG
lassFish(OSGiGlassFishRuntime.java:90)
        at com.sun.enterprise.glassfish.bootstrap.GlassFishMain$Launcher.launch(
GlassFishMain.java:113)
        ... 6 more
Caused by: org.glassfish.config.support.DomainXmlPreParser$DomainXmlPreParserExc
eption: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[53,149]
Message: 1バイトのUTF-8シーケンスのバイト1が無効です。
        at org.glassfish.config.support.DomainXmlPreParser.<init>(DomainXmlPrePa
rser.java:105)
        at org.glassfish.config.support.InstanceReaderFilter.<init>(InstanceRead
erFilter.java:72)
        at org.glassfish.config.support.DomainXml.parseDomainXml(DomainXml.java:
246)
        ... 16 more
Caused by: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[53,149]

Message: 1バイトのUTF-8シーケンスのバイト1が無効です。
        at com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl.next(XMLS
treamReaderImpl.java:599)
        at org.glassfish.config.support.DomainXmlPreParser.parse(DomainXmlPrePar
ser.java:140)
        at org.glassfish.config.support.DomainXmlPreParser.<init>(DomainXmlPrePa
rser.java:97)
        ... 18 more
````

## Fix
The issue is resolved by adding the character set encoding to the stream reader used in DomainXmlPreParser.

````
diff --git "a/C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\TortoiseGit\\DomainXmlPreParser-23e8e06.000.java" "b/C:\\gf5\\glassfish\\main\\nucleus\\admin\\config-api\\src\\main\\java\\org\\glassfish\\config\\support\\DomainXmlPreParser.java"
index 482ff7a332..1d2b74608d 100644
--- "a/C:\\Users\\ADMINI~1\\AppData\\Local\\Temp\\TortoiseGit\\DomainXmlPreParser-23e8e06.000.java"
+++ "b/C:\\gf5\\glassfish\\main\\nucleus\\admin\\config-api\\src\\main\\java\\org\\glassfish\\config\\support\\DomainXmlPreParser.java"
@@ -46,6 +46,7 @@ import java.io.*;
 import java.io.InputStream;
 import java.net.*;
 import java.net.URL;
+import java.nio.charset.Charset;
 import java.util.*;
 import javax.xml.stream.XMLInputFactory;
 import javax.xml.stream.XMLStreamException;
@@ -93,7 +94,8 @@ class DomainXmlPreParser {
         try {
             instanceName = instanceNameIn;
             in = domainXml.openStream();
-            reader = xif.createXMLStreamReader(domainXml.toExternalForm(), in);
+            InputStreamReader isr = new InputStreamReader(in, Charset.defaultCharset().toString());
+            reader = xif.createXMLStreamReader(domainXml.toExternalForm(), isr);
             parse();
             postProcess();
             validate();
````

### PR DESCRIPTION
**Pull request by mukumawa**
…ered in description field

Fixes #21893

### CODE REVIEW

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

File: nucleus/admin/config-api/src/main/java/org/glassfish/config/support/DomainXmlPreParser.java [hunk 1]
```diff
@@ -93,7 +94,8 @@ private DomainXmlPreParserException(String s) {
         try {
             instanceName = instanceNameIn;
             in = domainXml.openStream();
-            reader = xif.createXMLStreamReader(domainXml.toExternalForm(), in);
+            InputStreamReader isr = new InputStreamReader(in, Charset.defaultCharset().toString());
```
Comment (yaminikb) [inline nucleus/admin/config-api/src/main/java/org/glassfish/config/support/DomainXmlPreParser.java]: You need to close the stream reader.

**Review by yaminikb — CHANGES_REQUESTED** [review]

File: nucleus/admin/config-api/src/main/java/org/glassfish/config/support/DomainXmlPreParser.java (hunk 1 above)
Comment (mukumawa) [inline nucleus/admin/config-api/src/main/java/org/glassfish/config/support/DomainXmlPreParser.java]: Done

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

File: nucleus/admin/config-api/src/main/java/org/glassfish/config/support/DomainXmlPreParser.java [hunk 2]
```diff
@@ -105,6 +108,13 @@ private DomainXmlPreParserException(String s) {
             throw new DomainXmlPreParserException(e2);
         }
         finally {
+            try {
```
Comment (yaminikb) [inline nucleus/admin/config-api/src/main/java/org/glassfish/config/support/DomainXmlPreParser.java]: You can instead call cleanup().

**Review by yaminikb — CHANGES_REQUESTED** [review]

**Comment by mukumawa** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]

**Pull request merged**